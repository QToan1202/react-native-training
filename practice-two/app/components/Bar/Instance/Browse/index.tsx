import { useCallback, useMemo } from 'react'
import { XStack, YStack } from 'tamagui'
import { NativeStackHeaderProps } from '@react-navigation/native-stack'

import { BAR } from '@constants'
import Button, { ButtonProps } from '@components/Button'
import Search from '@components/Search'
import IconButton from '@components/IconButton'

import Bar from '@components/Bar'
import { IIconList } from '@types'

import styles from './styles'

const BrowseBar = ({ navigation, options, route }: NativeStackHeaderProps) => {
  // eslint-disable-next-line react-hooks/exhaustive-deps
  const handleNavigateToCart = useCallback(() => navigation.navigate('Cart'), [])
  // eslint-disable-next-line react-hooks/exhaustive-deps
  const handleNavigateToWishlist = useCallback(() => navigation.navigate('Wishlist'), [])
  const handlePress = () => {
    throw new Error('Function not implemented.')
  }
  const renderButton = useCallback(
    ({ title, leftIcon, onPress }: ButtonProps) => (
      <Button
        title={title}
        variant="tertiary"
        shrink
        style={styles.btn}
        leftIcon={leftIcon}
        onPress={onPress}
      />
    ),
    []
  )
  const IconList = useMemo(
    () =>
      BAR.HOME.map(({ label, ...rest }: IIconList) => (
        <IconButton
          key={label}
          onPress={label === 'cart' ? handleNavigateToCart : handleNavigateToWishlist}
          {...rest}
        />
      )),
    [handleNavigateToCart, handleNavigateToWishlist]
  )

  return (
    <Bar
      title={options.headerTitle?.toString() || route.name}
      IconList={IconList}
      align="space-between"
    >
      <YStack flex={1} space={28}>
        <Search placeholder="Search Product" />
        <XStack space={4} alignItems="center">
          {renderButton({
            title: 'sort by',
            leftIcon: require('@assets/sort.png'),
            onPress: handlePress,
          })}
          {renderButton({
            title: 'location',
            leftIcon: require('@assets/maps.png'),
            onPress: handlePress,
          })}
          {renderButton({
            title: 'category',
            leftIcon: require('@assets/category.png'),
            onPress: handlePress,
          })}
        </XStack>
      </YStack>
    </Bar>
  )
}

export default BrowseBar
