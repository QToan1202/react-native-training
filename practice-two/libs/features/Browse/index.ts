export * from './constants'
export * from './screens'
export { default as BrowseStack } from './navigation'
