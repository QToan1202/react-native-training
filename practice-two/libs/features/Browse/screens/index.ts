export { default as Browse } from './Browse'
export { default as CategoryDetail } from './Category'
