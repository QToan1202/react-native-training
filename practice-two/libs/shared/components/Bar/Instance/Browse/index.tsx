import { memo, useCallback, useMemo } from 'react'
import isEqual from 'react-fast-compare'
import { XStack, YStack, getTokenValue } from 'tamagui'
import { NativeStackHeaderProps } from '@react-navigation/native-stack'

import LoveIcon from '@assets/love.svg'
import CartIcon from '@assets/cart.svg'

import Button, { ButtonProps } from '../../../Button'
import Search from '../../../Search'
import IconButton from '../../../IconButton'
import Bar from '../../index'
import styles from './styles'

const BrowseBar = ({ navigation, options, route }: NativeStackHeaderProps) => {
  const handleNavigateToCart = useCallback(
    () => navigation.navigate('HomeStack', { screen: 'CheckoutStack', params: { screen: 'Cart' } }),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    []
  )
  const handleNavigateToWishlist = useCallback(
    () =>
      navigation.navigate('HomeStack', { screen: 'CheckoutStack', params: { screen: 'Wishlist' } }),
    // eslint-disable-next-line react-hooks/exhaustive-deps
    []
  )
  const handlePress = () => {
    throw new Error('Function not implemented.')
  }
  const renderButton = useCallback(
    ({ title, leftIcon, onPress }: ButtonProps) => (
      <Button
        title={title}
        variant="tertiary"
        shrink
        style={styles.btn}
        leftIcon={leftIcon}
        onPress={onPress}
      />
    ),
    []
  )
  const IconList = useMemo(
    () => (
      <>
        <IconButton noBackground onPress={handleNavigateToWishlist}>
          <LoveIcon fill={getTokenValue('$color.white')} />
        </IconButton>
        <IconButton noBackground onPress={handleNavigateToCart}>
          <CartIcon fill={getTokenValue('$color.white')} />
        </IconButton>
      </>
    ),
    [handleNavigateToCart, handleNavigateToWishlist]
  )

  return (
    <Bar
      title={options.headerTitle?.toString() || route.name}
      IconList={IconList}
      align="space-between"
    >
      <YStack flex={1} space={28}>
        <Search placeholder="Search Product" />
        <XStack space={4} alignItems="center">
          {renderButton({
            title: 'sort by',
            leftIcon: require('@assets/sort.png'),
            onPress: handlePress,
          })}
          {renderButton({
            title: 'location',
            leftIcon: require('@assets/maps.png'),
            onPress: handlePress,
          })}
          {renderButton({
            title: 'category',
            leftIcon: require('@assets/category.png'),
            onPress: handlePress,
          })}
        </XStack>
      </YStack>
    </Bar>
  )
}

export default memo(BrowseBar, isEqual)
