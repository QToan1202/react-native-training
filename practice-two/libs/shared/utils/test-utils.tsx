import { render, RenderOptions } from '@testing-library/react-native'
import React, { ReactElement } from 'react'
import { NavigationContainer } from '@react-navigation/native'
import { SafeAreaProvider } from 'react-native-safe-area-context'

const AllTheProviders = ({ children }: { children: ReactElement }) => {
  return (
    <SafeAreaProvider>
      <NavigationContainer>{children}</NavigationContainer>
    </SafeAreaProvider>
  )
}

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react-native'

export { customRender as render }
