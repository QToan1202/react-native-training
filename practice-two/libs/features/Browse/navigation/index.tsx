import { NativeStackHeaderProps, createNativeStackNavigator } from '@react-navigation/native-stack'

import { RootStackParamList } from '@practice-two/shared/types'
import { Feature } from '@practice-two/shared/hocs'
import { BrowseBar, CategoryBar } from '@practice-two/shared/components'
import { CustomHeader } from '@practice-two/shared/utils'

const Stack = createNativeStackNavigator<RootStackParamList>()

const BrowseStack = () => (
  <Feature feat="browse">
    <Stack.Navigator>
      <Stack.Screen
        name="CategoryDetail"
        getComponent={() => require('../screens').CategoryDetail}
        options={{
          header: (props: NativeStackHeaderProps) => CustomHeader(CategoryBar, props),
        }}
      />
      <Stack.Screen
        name="Browse"
        getComponent={() => require('../screens').Browse}
        options={{
          header: (props: NativeStackHeaderProps) => CustomHeader(BrowseBar, props),
        }}
      />
    </Stack.Navigator>
  </Feature>
)

export default BrowseStack
