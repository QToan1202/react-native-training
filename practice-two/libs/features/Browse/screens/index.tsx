import { useCallback, useMemo } from 'react'
import { Spinner, YStack } from 'tamagui'
import { FlatList, ListRenderItem } from 'react-native'
import { CompositeScreenProps } from '@react-navigation/native'
import { BottomTabScreenProps } from '@react-navigation/bottom-tabs'
import { NativeStackScreenProps } from '@react-navigation/native-stack'

import {
  ProductCard,
  Heading,
  Paragraph,
  IProduct,
  productListStyles,
  useGetProducts,
} from '@practice-two/shared'
import { TabParamsList } from 'libs/shared/navigation/Tab'
import { RootStackParamList } from 'libs/shared/navigation/Stack'

export type BrowseScreenProps = CompositeScreenProps<
  BottomTabScreenProps<TabParamsList, 'BrowseTab'>,
  NativeStackScreenProps<RootStackParamList>
>

const Browse = ({ navigation }: BrowseScreenProps) => {
  const { data: products, isSuccess, isLoading } = useGetProducts(process.env.PRODUCT_ENDPOINT)

  const handleMoveToProduct = useCallback((id: string) => {
    navigation.navigate('BrowseStack', { screen: 'ProductDetail', params: { id } })
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])
  const renderProductItem: ListRenderItem<IProduct> = useCallback(
    ({ item }) => {
      if (!isSuccess) return null
      const { id, img, price, discountPrice, name, store } = item

      return (
        <ProductCard
          id={String(id)}
          img={{ uri: img }}
          price={price}
          discountPrice={discountPrice}
          title={name}
          avatar={{ uri: store.avatar }}
          storeName={store.name}
          onPressCard={handleMoveToProduct}
        />
      )
    },
    [isSuccess, handleMoveToProduct]
  )
  const Item = useMemo(() => {
    if (isLoading) return <Spinner size="large" color="$color.primary" />
    if (!isSuccess) return null
    if (!products.length)
      return (
        <YStack paddingVertical="$space.4" paddingHorizontal="$space.10">
          <Heading
            content="List empty!!!!"
            color="$color.dark_50"
            fontWeight="$3"
            fontSize="$3"
            textAlign="center"
          />
          <Paragraph
            content="Don't have any products"
            color="$color.gray_50"
            fontSize="$2"
            fontWeight="$2"
            textAlign="center"
          />
        </YStack>
      )

    return (
      <FlatList
        keyExtractor={({ id }: IProduct): string => String(id)}
        data={products}
        renderItem={renderProductItem}
        numColumns={2}
        showsHorizontalScrollIndicator={false}
        style={productListStyles.container}
        contentContainerStyle={productListStyles.item}
        columnWrapperStyle={productListStyles.column}
      />
    )
  }, [isLoading, isSuccess, products, renderProductItem])

  return (
    <YStack flex={1} backgroundColor="$color.bg_layer">
      {Item}
    </YStack>
  )
}

export default Browse
