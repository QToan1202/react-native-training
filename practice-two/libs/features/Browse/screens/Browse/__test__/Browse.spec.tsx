import { render } from 'test-utils'

import Browse from '../index'

describe('Testing Category Detail screen', () => {
  it('Render screen successfully', () => {
    const component = render(<Browse />).toJSON()

    expect(component).toMatchSnapshot()
  })
})
