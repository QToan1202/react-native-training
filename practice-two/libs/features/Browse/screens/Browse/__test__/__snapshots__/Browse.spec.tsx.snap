// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Category Detail screen Render screen successfully 1`] = `
<RCTScrollView
  contentContainerStyle={
    [
      {
        "alignSelf": "center",
        "paddingVertical": 30,
        "rowGap": 10,
      },
      undefined,
    ]
  }
  data={
    [
      {
        "avatar": {
          "testUri": "../../../assets/avatar.png",
        },
        "id": "1",
        "img": {
          "testUri": "../../../assets/product/coca.png",
        },
        "price": 222.6,
        "storeName": "tradly",
        "title": "coca cola",
      },
      {
        "avatar": {
          "testUri": "../../../assets/avatar.png",
        },
        "id": "2",
        "img": {
          "testUri": "../../../assets/product/coca.png",
        },
        "price": 222.6,
        "storeName": "tradly",
        "title": "coca cola",
      },
      {
        "avatar": {
          "testUri": "../../../assets/avatar.png",
        },
        "discountPrice": 12,
        "id": "3",
        "img": {
          "testUri": "../../../assets/product/coca.png",
        },
        "price": 222.6,
        "storeName": "tradly",
        "title": "coca cola",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsHorizontalScrollIndicator={false}
  stickyHeaderIndices={[]}
  style={
    {
      "backgroundColor": "#fff",
    }
  }
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            [
              {
                "columnGap": 10,
                "justifyContent": "flex-start",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignSelf": "flex-start",
              "backgroundColor": "#fff",
              "borderColor": "#0000001a",
              "borderRadius": 10,
              "borderStyle": "solid",
              "borderWidth": 1,
              "opacity": 1,
              "overflow": "hidden",
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../assets/product/coca.png",
              }
            }
            style={
              {
                "height": 128,
                "width": 160,
              }
            }
          />
          <View
            style={
              {
                "padding": 12,
                "rowGap": 16,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  {
                    "color": "#000",
                    "fontFamily": "Montserrat_500Medium",
                    "maxWidth": 140,
                    "textTransform": "capitalize",
                  },
                ]
              }
            >
              coca cola
            </Text>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "columnGap": 4,
                    "flexDirection": "row",
                  },
                  {
                    "justifyContent": "space-between",
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "columnGap": 4,
                      "flexDirection": "row",
                    },
                    {
                      "columnGap": 10,
                    },
                    undefined,
                  ]
                }
              >
                <Image
                  accessibilityLabel="tradly-avatar"
                  source={
                    {
                      "testUri": "../../../assets/avatar.png",
                    }
                  }
                  style={
                    [
                      {
                        "borderRadius": 50,
                      },
                      {
                        "height": 20,
                        "width": 20,
                      },
                    ]
                  }
                />
                <Text
                  numberOfLines={1}
                  style={
                    [
                      {
                        "color": "#fff",
                        "fontFamily": "Montserrat_400Regular",
                      },
                      {
                        "fontSize": 14,
                      },
                      {
                        "color": "#4f4f4f",
                        "fontFamily": "Montserrat_500Medium",
                        "textTransform": "capitalize",
                      },
                    ]
                  }
                >
                  tradly
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "columnGap": 4,
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    [
                      {
                        "color": "#fff",
                        "fontFamily": "Montserrat_400Regular",
                      },
                      {
                        "fontSize": 14,
                      },
                      {
                        "color": "#33907C",
                        "fontFamily": "Montserrat_600SemiBold",
                        "maxWidth": 35,
                      },
                    ]
                  }
                >
                  $223
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignSelf": "flex-start",
              "backgroundColor": "#fff",
              "borderColor": "#0000001a",
              "borderRadius": 10,
              "borderStyle": "solid",
              "borderWidth": 1,
              "opacity": 1,
              "overflow": "hidden",
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../assets/product/coca.png",
              }
            }
            style={
              {
                "height": 128,
                "width": 160,
              }
            }
          />
          <View
            style={
              {
                "padding": 12,
                "rowGap": 16,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  {
                    "color": "#000",
                    "fontFamily": "Montserrat_500Medium",
                    "maxWidth": 140,
                    "textTransform": "capitalize",
                  },
                ]
              }
            >
              coca cola
            </Text>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "columnGap": 4,
                    "flexDirection": "row",
                  },
                  {
                    "justifyContent": "space-between",
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "columnGap": 4,
                      "flexDirection": "row",
                    },
                    {
                      "columnGap": 10,
                    },
                    undefined,
                  ]
                }
              >
                <Image
                  accessibilityLabel="tradly-avatar"
                  source={
                    {
                      "testUri": "../../../assets/avatar.png",
                    }
                  }
                  style={
                    [
                      {
                        "borderRadius": 50,
                      },
                      {
                        "height": 20,
                        "width": 20,
                      },
                    ]
                  }
                />
                <Text
                  numberOfLines={1}
                  style={
                    [
                      {
                        "color": "#fff",
                        "fontFamily": "Montserrat_400Regular",
                      },
                      {
                        "fontSize": 14,
                      },
                      {
                        "color": "#4f4f4f",
                        "fontFamily": "Montserrat_500Medium",
                        "textTransform": "capitalize",
                      },
                    ]
                  }
                >
                  tradly
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "columnGap": 4,
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    [
                      {
                        "color": "#fff",
                        "fontFamily": "Montserrat_400Regular",
                      },
                      {
                        "fontSize": 14,
                      },
                      {
                        "color": "#33907C",
                        "fontFamily": "Montserrat_600SemiBold",
                        "maxWidth": 35,
                      },
                    ]
                  }
                >
                  $223
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <View
        style={
          [
            {
              "flexDirection": "row",
            },
            [
              {
                "columnGap": 10,
                "justifyContent": "flex-start",
              },
              undefined,
            ],
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignSelf": "flex-start",
              "backgroundColor": "#fff",
              "borderColor": "#0000001a",
              "borderRadius": 10,
              "borderStyle": "solid",
              "borderWidth": 1,
              "opacity": 1,
              "overflow": "hidden",
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../assets/product/coca.png",
              }
            }
            style={
              {
                "height": 128,
                "width": 160,
              }
            }
          />
          <View
            style={
              {
                "padding": 12,
                "rowGap": 16,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                [
                  {
                    "color": "#fff",
                    "fontFamily": "Montserrat_400Regular",
                  },
                  {
                    "fontSize": 14,
                  },
                  {
                    "color": "#000",
                    "fontFamily": "Montserrat_500Medium",
                    "maxWidth": 140,
                    "textTransform": "capitalize",
                  },
                ]
              }
            >
              coca cola
            </Text>
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "columnGap": 4,
                    "flexDirection": "row",
                  },
                  {
                    "justifyContent": "space-between",
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "columnGap": 4,
                      "flexDirection": "row",
                    },
                    {
                      "columnGap": 10,
                    },
                    undefined,
                  ]
                }
              >
                <Image
                  accessibilityLabel="tradly-avatar"
                  source={
                    {
                      "testUri": "../../../assets/avatar.png",
                    }
                  }
                  style={
                    [
                      {
                        "borderRadius": 50,
                      },
                      {
                        "height": 20,
                        "width": 20,
                      },
                    ]
                  }
                />
                <Text
                  numberOfLines={1}
                  style={
                    [
                      {
                        "color": "#fff",
                        "fontFamily": "Montserrat_400Regular",
                      },
                      {
                        "fontSize": 14,
                      },
                      {
                        "color": "#4f4f4f",
                        "fontFamily": "Montserrat_500Medium",
                        "textTransform": "capitalize",
                      },
                    ]
                  }
                >
                  tradly
                </Text>
              </View>
              <View
                style={
                  {
                    "alignItems": "center",
                    "columnGap": 4,
                    "flexDirection": "row",
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    [
                      {
                        "color": "#fff",
                        "fontFamily": "Montserrat_400Regular",
                      },
                      {
                        "fontSize": 14,
                      },
                      {
                        "color": "#4f4f4f",
                        "fontSize": 10,
                        "maxWidth": 35,
                        "textDecorationLine": "line-through",
                      },
                    ]
                  }
                >
                  $223
                </Text>
                <Text
                  numberOfLines={1}
                  style={
                    [
                      {
                        "color": "#fff",
                        "fontFamily": "Montserrat_400Regular",
                      },
                      {
                        "fontSize": 14,
                      },
                      {
                        "color": "#33907C",
                        "fontFamily": "Montserrat_600SemiBold",
                        "maxWidth": 35,
                      },
                    ]
                  }
                >
                  $12
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
